frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -62,8 +92,60 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
   const removePlugins: boolean =
     !!consoleOperatorConfig && canPatchConsoleOperatorConfig && enabledPlugins.length > 0;
 
+  const subscriptionName = subscription?.spec?.name;
+  const subscriptionNamespace = subscription?.metadata?.namespace;
+
+  const isSubmitInProgress = React.useMemo(
+    () => operatorUninstallInProgress || operandsDeleteInProgress,
+    [operatorUninstallInProgress, operandsDeleteInProgress],
+  );
+
+  const isSubmitFinished = React.useMemo(
+    () =>
+      !isSubmitInProgress &&
+      (isDeleteAllOperandsSelected ? operandsDeleteFinished : true) &&
+      operatorUninstallFinished,
+    [
+      isSubmitInProgress,
+      isDeleteAllOperandsSelected,
+      operandsDeleteFinished,
+      operatorUninstallFinished,
+    ],
+  );
+
+  const areSubmitErrors = React.useMemo(
+    () =>
+      areOperandDeletionErrors(operandDeletionErrors) || operatorUninstallErrorMessage.length > 0,
+    [operandDeletionErrors, operatorUninstallErrorMessage],
+  );
+
+  const [operands, operandsLoaded, operandsLoadedErrorMessage] = useOperands(
+    subscriptionName,
+    subscriptionNamespace,
+  );
+
+  const closeAndRedirect = React.useCallback(() => {
+    close();
+    if (
```

Comment:
This looks quite hacky to me. Would suggest to add at least an explanatory comment.

Reply:
Hi, is `closeAndRedirect` not descriptive enough? :-)  This is the original code, I just moved it to a function.  -thanks

Reply:
It's probably better to make this a behavior a prop passed to the modal instead of a pattern on the path, but we can leave this since for now since it's not new.